airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java

```diff
@@ -108,40 +113,74 @@ public Map<StreamDescriptor, MemoryBoundedLinkedBlockingQueue<AirbyteMessage>> g
       return new HashMap<>(buffers);
     }
 
-    public MemoryBoundedLinkedBlockingQueue<AirbyteMessage> getBuffer(final StreamDescriptor streamDescriptor) {
+    private MemoryBoundedLinkedBlockingQueue<AirbyteMessage> getBuffer(final StreamDescriptor streamDescriptor) {
       return buffers.get(streamDescriptor);
     }
 
-    public long getTotalGlobalQueueSizeInMb() {
+    public long getTotalGlobalQueueSizeBytes() {
       return buffers.values().stream().map(MemoryBoundedLinkedBlockingQueue::getCurrentMemoryUsage).mapToLong(Long::longValue).sum();
     }
 
-    public long getQueueSizeInMb(final StreamDescriptor streamDescriptor) {
+    public long getQueueSizeInRecords(final StreamDescriptor streamDescriptor) {
+      return getBuffer(streamDescriptor).size();
+    }
+
+    public long getQueueSizeBytes(final StreamDescriptor streamDescriptor) {
       return getBuffer(streamDescriptor).getCurrentMemoryUsage();
     }
 
     public Optional<Instant> getTimeOfLastRecord(final StreamDescriptor streamDescriptor) {
       return getBuffer(streamDescriptor).getTimeOfLastMessage();
     }
 
-    public Batch take(final StreamDescriptor streamDescriptor, long bytesToRead) {
-      var queue = buffers.get(streamDescriptor);
-      return new Batch(List.of());
-    }
+    public Batch take(final StreamDescriptor streamDescriptor, final long bytesToRead) {
+      final var queue = buffers.get(streamDescriptor);
 
-    class Batch implements AutoCloseable {
+      final AtomicLong bytesRead = new AtomicLong();
```

Comment:
does this need to be an atomic long? this looks like one thread

Reply:
java doesn't like accessing not final things from within a stream. :P

Reply:
😢